### PR TITLE
React Data Fetching: ISS Tracker

### PR DESCRIPTION
**Pull request by PReichetanz**
### 🔎 Review

Review via [🔗 **CodeSandbox**](https://codesandbox.io/s/github/neuefische/web-exercises/tree/react-data-fetching_iss/sessions/react-data-fetching/iss-tracker?file=/README.md) or locally by running this command in your Terminal:

```
npx ghcd@latest neuefische/web-exercises/tree/react-data-fetching_iss/sessions/react-data-fetching/iss-tracker -i
```

### ✨ Copyable Markdown for Curriculum

````md
Solve this challenge on
[🔗 **CodeSandbox**](https://codesandbox.io/s/github/neuefische/web-exercises/tree/main/sessions/react-data-fetching/iss-tracker?file=/README.md)
or locally by running this command in your Terminal:

```
npx ghcd@latest neuefische/web-exercises/tree/main/sessions/react-data-fetching/iss-tracker -i
```
````

refers to neuefische/web-curriculum-new-format#216

### CODE REVIEW

File: sessions/react-data-fetching/iss-tracker/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  - the `getISSCoords` function,
+  - the `useEffect`,
+  - for now: the `onRefresh` prop passed to the `Controls` component (you will adapt this in a minute).
+
+- Implement `SWR` to fetch the same data.
+
+  - `SWR` is already installed, so you just have to import it.
+  - Create a loading and an error state as well.
+  - Remember to adapt the `fetcher` function to create a proper `error` state [as explained in the SWR docs](https://swr.vercel.app/docs/error-handling#status-code-and-error-object).
+  - Note: `SWR` needs you to destructure a `data` object, but the `Map` and `Controls` component need a `coords` object passed to their props; how can you simply fix this naming issue without touching the props?
```
Comment (aclaes) [inline sessions/react-data-fetching/iss-tracker/README.md:25]:
```suggestion
  - Note: `SWR` needs you to destructure a `data` object, but the `Map` and `Controls` component need `longitude` and `latitude` as separate props; how can you simply pass the coordinates from `data` without touching the props?
```

**Review by aclaes — CHANGES_REQUESTED** [review]
Good challenge! I have a few improvements ideas

File: sessions/react-data-fetching/iss-tracker/README.md (hunk 1 above)
Comment (PReichetanz) [reply sessions/react-data-fetching/iss-tracker/README.md]: We have found a good compromise.

File: sessions/react-data-fetching/iss-tracker/components/ISSTracker/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    latitude: 0,
+  });
+
+  async function getISSCoords() {
+    try {
+      const response = await fetch(URL);
+      if (response.ok) {
+        const data = await response.json();
+        setCoords({ longitude: data.longitude, latitude: data.latitude });
+        console.log(data);
```
Comment (aclaes) [inline sessions/react-data-fetching/iss-tracker/components/ISSTracker/index.js:19]:
```suggestion
```

File: sessions/react-data-fetching/iss-tracker/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+### Fetching on load
+
+- Comment out all unnecessary code:
+
+  - the `coords` state,
+  - the `getISSCoords` function,
+  - the `useEffect`,
+  - for now: the `onRefresh` prop passed to the `Controls` component (you will adapt this in a minute).
+
+- Implement `SWR` to fetch the same data.
```
Comment (aclaes) [inline sessions/react-data-fetching/iss-tracker/README.md:20]:
```suggestion
- Implement fetching of the same data with `SWR`.
```

**Review by aclaes — APPROVED** [review]
Very good, approve 🎉 Just two little improvements

**Pull request merged**